### PR TITLE
Dialyzer spec fixes

### LINKED ISSUE #22: Incorrect spec for jobs_server:add_queue
Trying to add a queue with `jobs:add_queue(Name, [{regulators, ...}])` throws a warning in dialyzer:

`The call jobs:add_queue(Name, Props::[{'regulators',[any(),...]},...]) will never return since it differs in the 2nd argument from the success typing arguments: (any(), [{'config',atom() | [atom() | [any()] | char()]} | {'counters',[{_,_}]} | {'group_rates',[{_,_}]} | {'interval',integer()} | {'queues',[{_,_} | {_,_,_}]}])`

The `jobs_server:add_queue` spec is defined here:
https://github.com/uwiger/jobs/blob/a04e8d4e03d9d231cf343d5aef24431d39ec037d/src/jobs_server.erl#L148

And the `option()` type doesn't allow `regulators` as a top level option:
https://github.com/uwiger/jobs/blob/a04e8d4e03d9d231cf343d5aef24431d39ec037d/include/jobs.hrl#L32-L45

From @uwiger on Slack:

> the type spec for `add_queue()` is wrong. From a quick look at the code, I think it should be `-spec add_queue(queue_name(), q_opts()) -> ok.`
> … and `q_opts()` could well be improved to be more precise.

### PR DESCRIPTION
**Pull request by jwheare**
I fleshed out the types and fixed various spec issues, reducing the internal dialyzer warnings to just two about: `Callback info about the jobs_sampler behaviour is not available`

I also tidied up mixed space/tab use in the hrl file. No code changes were made and tests still pass.

This should resolve #22

### CODE REVIEW

File: src/jobs_server.erl [hunk 1]
```diff
@@ -89,7 +89,7 @@ ask() ->
     ask(default).
 
 
--spec ask(job_class()) -> {ok, reg_obj()} | {error, rejected | timeout}.
+-spec ask(job_class()) -> {ok, counter()} | {error, rejected | timeout}.
```
Comment (uwiger) [inline src/jobs_server.erl]: Actually, the return type is not `{ok, counter()}`, but I can understand if dialyzer accepts it.
It should, IMHO, be `{ok, reg_obj()}`, where `reg_obj() :: {reference(), [{info, any()} | {counters, [counter()]}]}`, and where `counters() :: {#cr{}, non_neg_integer()}`

Several compounding type errors in the old code, that is.

**Comment by uwiger** [comment]
Very good. But one thing that caught my eye. I added a comment about it.

**Comment by jwheare** [comment]
Made the suggested changes and fixed the final dialyzer warnings.

**Comment by uwiger** [comment]
Merged. Sorry for the delay, and thanks.

**Comment by uwiger** [comment]
Version 0.7.1 pushed and published on hex

**Pull request merged**